reactive/kubernetes_master.py

```diff
@@ -1084,6 +1100,10 @@ def configure_cdk_addons():
     enable_gcp = 'false'
     enable_openstack = str(is_flag_set('endpoint.openstack.ready')).lower()
     openstack = endpoint_from_flag('endpoint.openstack.ready')
+    if is_state('kubernetes-master.cluster-name.new'):
+        cluster_name = leader_get('cluster_tag')
+    elif is_state('kubernetes-master.cluster-name.legacy'):
```

Comment:
Do we really need two flags here?  Why not just react to `if .new: foo; else: bar`?

Reply:
We need two flags because we need to handle four cases: a fresh install, an upgrade of old code that hasn't called the `upgrade` action with the flag yet, an upgrade of old code which has already called the action, and an upgrade of new code.  I couldn't figure out another way to distinguish between those with a single flag.  (Some of those cases could maybe be combined, but I still ended up needing both flags.)